### ISSUE TITLE
"!" in single quoted strings lead to evaluation

### ISSUE THREAD

**Issue by anntzer** [robbyrussell/oh-my-zsh#6052]
Without any zsh config,
```
$ echo '!foo'
```
just prints `!foo`, as expected.

With a clean oh-my-zsh install, using the given zshrc template (https://github.com/robbyrussell/oh-my-zsh/blob/master/templates/zshrc.zsh-template), it prints `foo'!foo`.

**Comment by BenGitsCode** [comment]
Strange. I thought I had a pretty clean zshrc but my terminal just prints `!foo`

**Comment by anntzer** [comment]
Looks like a bug in konsole, introduced at some point between 16.12.3 and 17.04.0, which leads to misinterpretation of
```
      print -Pn "\e]2;$2:q\a" # set window name
```
in termsupport.zsh.

Unfortunately I have trouble creating a minimal example to report this upstream, any suggestions?

**Comment by anntzer** [comment]
This seems fixed as of konsole 17.04.1.
